search/binary_search1.cpp

```diff
@@ -0,0 +1,239 @@
+/**
+ * @file
+ * @brief [Binary search
+ * algorithm](https://en.wikipedia.org/wiki/Binary_search_algorithm)
+ * @author [Mohammad Golzar](https://github.com/m-golzar)
+ */
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <string>
+#include <array>
+#include <cassert>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```

Reply:
Thank you @Panquesito7 for your suggestions, I will modify the code.